### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by joefitter**
Session IO functionality. Added mixin for reading/writing to the sessionModel. Taken originally from hof-form-wizard/lib/controller.js and repackaged as a mixin

### CODE REVIEW

**Comment by lennym** [comment]
Is there a good argument for this being separate from `hof-form-controller`?

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+extends:
+- "homeoffice/config/default"
+
+env:
+  es6: true
+
+rules:
+  no-underscore-dangle: 0
+  object-curly-spacing: 0
```
Comment (lennym) [inline .eslintrc:9]: If this is a rule we want to disable then PR `eslint-config-homeoffice`

**Review by lennym — APPROVED** [review]

**Comment by joefitter** [comment]
It _could_ be part of that repo, but then again so could hooks. I guess it depends on the level of ~~abstraction~~ separation you want. This repo will be required by and mixed in to the main export from hof-form-controller

**Comment by lennym** [comment]
I think it's probably fine as it is then. As long as everything is encapsulated in the end so it's opaque to the implementer.

**Comment by lennym** [comment]
All I would add is something at the `configure` phase to ensure `sessionModel` actually exists.

**Comment by joefitter** [comment]
That is a good point, I'll add to this PR

**Comment by lennym** [comment]
I'm torn between just throwing an error, or falling back to a basic in-memory session store backed model with a big warning saying to implement a proper session model.

Tending towards the former for safety and simplicity.

**Comment by joefitter** [comment]
former wins :)

**Comment by lennym** [comment]
I was just thinking "I'm going to really annoy him by suggesting adding something to the readme about session middleware" but you're one step ahead of me.

APPROVED!

**Pull request merged**